### DISCUSSION TITLE
supabase-js: validate the `ANON_KEY`

### DISCUSSION THREAD

**Discussion by davidkhala** [supabase/supabase#22244] [Questions] [answered]
Before intent to give command, is there any way we can validate the ANON_KEY only? If not possible, any operation we can have it in dry-run mode?

**Comment by GaryAustin1** [comment]
Can you explain what you are trying to do?

**Reply by davidkhala** [reply]
As mentioned, just to validate the ANON_KEY only.

Typically, we establish a paradigm that for a nodejs backend client, it always has a `connect` function to access data provider. the `connect` function will return a `connection` object, which can be a session or a long-live db connection got from connection pool. If there is any credentail needed for establish a `connection`, it should be validated within the `connect` function instead of being lazy that client will get rejected in using `connection` afterwards.

Knowing supabase-js is not a db driver, but we hope it can still be adapted to this paradigm.

**Reply by GaryAustin1** [reply, marked as answer]
The REST API's don't have connections.  Each request is independent.  You could I guess make a call to a known table and select no/minimal data or an rpc call if you want.
